x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import Boom from '@hapi/boom';
+import type { SearchQuery } from '@kbn/content-management-plugin/common';
+import type { ContentStorage, StorageContext } from '@kbn/content-management-plugin/server';
+import type {
+  SavedObject,
+  SavedObjectReference,
+  SavedObjectsFindOptions,
+} from '@kbn/core-saved-objects-api-server';
+
+import { CONTENT_ID } from '../../common/content_management';
+import { cmServicesDefinition } from '../../common/content_management/cm_services';
+import type {
+  MapItem,
+  PartialMapItem,
+  MapContentType,
+  MapAttributes,
+  MapGetOut,
+  MapCreateIn,
+  MapCreateOut,
+  CreateOptions,
+  MapUpdateIn,
+  MapUpdateOut,
+  UpdateOptions,
+  MapDeleteOut,
+  MapSearchOptions,
+  MapSearchOut,
+} from '../../common/content_management';
+
+const savedObjectClientFromRequest = async (ctx: StorageContext) => {
+  if (!ctx.requestHandlerContext) {
+    throw new Error('Storage context.requestHandlerContext missing.');
+  }
+
+  const { savedObjects } = await ctx.requestHandlerContext.core;
+  return savedObjects.client;
```

Comment:
noob question:

Is the fact that CM provides access to the SO-client, primarily sugar for CM-clients? Or is there a specific reason that this needs to be passed along with every request?

Reply:
The `savedObjects` passed in the request is the one scoped to the current user and thus has security baked in.